lib/hijacker.rb

```diff
@@ -75,16 +84,33 @@ def self.connect(target_name, sister_name = nil, options = {})
       reenable_query_caching
 
       run_after_hijack_callback
+
     rescue
+      if database
+        conn_config = connection_config(database, slave_connection = false, {check_responsiveness: false})
```

Comment:
Under normal circumstances, it's in `connection_config` that a check is made to see if the host is available or not.  In this case, we only want to get the configuration without the intent to actually make a connection, hence the option to not check for db host responsiveness.